src/test/architecture/test_architecture.py

```diff
@@ -1019,3 +1019,140 @@ def test_rename_path(self, mocker):
         assert set(arch.communications) == set(arch_expected.communications)
         assert set(arch.paths) == set(arch_expected.paths)
         assert set(arch.executors) == set(arch_expected.executors)
+
+
+class TestDiff:
+
+    def test_diff_node_names(self, mocker):
+        architecture_mock1 = mocker.Mock(spec=Architecture)
+        architecture_mock2 = mocker.Mock(spec=Architecture)
+        mocker.patch.object(architecture_mock1,
+                            'node_names',
+                            ('drive_node', 'actuator_dummy_node'))
+        mocker.patch.object(architecture_mock2,
+                            'node_names',
+                            ('drive_node', 'filter_node', 'message_driven_node'))
+        name1, name2 = Architecture.diff_node_names(architecture_mock1, architecture_mock2)
+        assert name1 == ('actuator_dummy_node',)
+        assert name2 == ('filter_node', 'message_driven_node') or \
+               name2 == ('message_driven_node', 'filter_node')
+
+        mocker.patch.object(architecture_mock1, 'node_names', ('drive_node',))
+        mocker.patch.object(architecture_mock2, 'node_names', ())
+        name1, name2 = Architecture.diff_node_names(architecture_mock1, architecture_mock2)
+        assert name1 == ('drive_node',)
+        assert name2 == ()
+
+        mocker.patch.object(architecture_mock1, 'node_names', ())
+        mocker.patch.object(architecture_mock2, 'node_names', ('drive_node', ))
+        name1, name2 = Architecture.diff_node_names(architecture_mock1, architecture_mock2)
+        assert name1 == ()
+        assert name2 == ('drive_node',)
+
+    def test_diff_node_topics(self, mocker):
+        architecture_mock1 = mocker.Mock(spec=Architecture)
+        architecture_mock2 = mocker.Mock(spec=Architecture)
+        mocker.patch.object(architecture_mock1, 'topic_names', ('topic1', 'topic2'))
+        mocker.patch.object(architecture_mock2, 'topic_names', ('topic1', 'topic3', 'topic4'))
+        topic_1, topic_2 = Architecture.diff_node_topics(architecture_mock1, architecture_mock2)
+        assert topic_1 == ('topic2',)
+        assert topic_2 == ('topic3', 'topic4') or \
+               topic_2 == ('topic4', 'topic3')
```

Comment:
In order to adapt to any order, how about the following implementation?

```suggestion
        assert sorted(topic_2) == sorted(('topic3', 'topic4'))
```